nemo/collections/common/callbacks/callbacks.py

```diff
@@ -30,3 +32,40 @@ def on_epoch_end(self, trainer, pl_module):
         curr_time = time.time()
         duration = curr_time - self.epoch_start
         trainer.logger.log_metrics({"epoch_time": duration}, step=trainer.global_step)
+
+
+class LogTrainValidLossCallback(Callback):
+    @rank_zero_only
+    def on_train_start(self, trainer, pl_module):
+        logging.info("Training started")
+
+    @rank_zero_only
+    def on_train_batch_end(self, trainer, pl_module, batch, batch_idx, dataloader_idx):
+        print_freq = trainer.row_log_interval
+        total_batches = trainer.num_training_batches
+        if 0 < print_freq < 1:
+            print_freq = int(total_batches * print_freq)
+        if batch_idx % print_freq == 0:
+            logging.info(
+                "Epoch: {}/{} batch: {}/{} train_loss: {:.3f} train_acc: {:.2f}".format(
+                    trainer.current_epoch + 1,
+                    trainer.max_epochs,
+                    batch_idx + 1,
+                    total_batches,
+                    pl_module.loss_value,
+                    pl_module.accuracy,
```

Comment:
Not all modules have `accuracy` so this callback will fail for a lot of models. Why not just read the log in its entirety and just print all of the values in the log? Cant we access the log at the end of train_batch_end?

Reply:
Unfortunately not yet, PTL is working on it